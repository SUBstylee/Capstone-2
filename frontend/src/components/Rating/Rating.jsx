import "./Rating.scss";

const Rating = ({ value, text, color }) => {
  return (
    <div className='Rating'>
      <span>
        <i
          style={{ color }}
          className={
            value >= 1
              ? "fa-solid fa-star"
              : value >= 0.5
              ? "fa-solid fa-star-half-stroke"
              : "fa-regular fa-star"
          }
        ></i>
      </span>
      <span>
        <i
          style={{ color }}
          className={
            value >= 2
              ? "fa-solid fa-star"
              : value >= 1.5
              ? "fa-solid fa-star-half-stroke"
              : "fa-regular fa-star"
          }
        ></i>
      </span>
      <span>
        <i
          style={{ color }}
          className={
            value >= 3
              ? "fa-solid fa-star"
              : value >= 2.5
              ? "fa-solid fa-star-half-stroke"
              : "fa-regular fa-star"
          }
        ></i>
      </span>
      <span>
        <i
          style={{ color }}
          className={
            value >= 4
              ? "fa-solid fa-star"
              : value >= 3.5
              ? "fa-solid fa-star-half-stroke"
              : "fa-regular fa-star"
          }
        ></i>
      </span>
      <span>
        <i
          style={{ color }}
          className={
            value >= 5
              ? "fa-solid fa-star"
              : value >= 4.5
              ? "fa-solid fa-star-half-stroke"
              : "fa-regular fa-star"
          }
        ></i>
      </span>
      <span className='ps-2'>{text && text}</span>
    </div>
  );
};

Rating.defaultProps = {
  color: "#EEBC1D",
};

export default Rating;
