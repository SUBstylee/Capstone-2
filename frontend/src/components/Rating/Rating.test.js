import { render } from '@testing-library/react';
import Rating from './Rating';
import { BrowserRouter } from 'react-router-dom';

describe('test', () => {
    let value, text;
    beforeEach(() => {
        (value = 5), (text = `1 review`)
    });
    test("renders Rating component without crashing", () => {
        const { getByText } = render(
            <BrowserRouter>
                <Rating value={value} text={text} />
            </BrowserRouter>
        );

        expect(getByText('1 review')).toBeInTheDocument();
    });

    test("matches snapshot", function () {
        const { asFragment } = render(
            <BrowserRouter>
                <Rating value={value} text={text} />
            </BrowserRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });

});