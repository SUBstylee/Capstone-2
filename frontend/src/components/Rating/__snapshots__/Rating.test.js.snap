// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Rating"
  >
    <span>
      <i
        class="fa-solid fa-star"
        style="color: rgb(238, 188, 29);"
      />
    </span>
    <span>
      <i
        class="fa-solid fa-star"
        style="color: rgb(238, 188, 29);"
      />
    </span>
    <span>
      <i
        class="fa-solid fa-star"
        style="color: rgb(238, 188, 29);"
      />
    </span>
    <span>
      <i
        class="fa-solid fa-star"
        style="color: rgb(238, 188, 29);"
      />
    </span>
    <span>
      <i
        class="fa-solid fa-star"
        style="color: rgb(238, 188, 29);"
      />
    </span>
    <span
      class="ps-2"
    >
      1 review
    </span>
  </div>
</DocumentFragment>
`;
